experiments/dsl/src/main/scala/org/apache/gearpump/streaming/dsl/StreamApp.scala

```diff
@@ -91,7 +88,7 @@ object StreamApp {
 class TimeReplayableProducer[T:ClassTag](timeReplayableSource: TimeReplayableSource, converter: Message => T, batchSize: Int) extends Traverse[T] {
   val source = timeReplayableSource
   override def foreach[U](fun: T => U): Unit = {
-    val list = source.pull(batchSize)
+    val list = source.read()
```

Comment:
I think it is better let caller instead of callee of ```read()``` to decide the max batch size as caller, in theory, has better understanding of the pressure it has.